rego-templates/embedded_test.go

```diff
@@ -0,0 +1,143 @@
+package rego_templates
+
+import (
+	"embed"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/aquasecurity/postee/v2/data"
+)
+
+//go:embed testdata/*.rego
+var embedTest embed.FS
+
+func TestPopulateTemplate(t *testing.T) {
+	expected := map[string]string{
+		"test.rego": `package test.rawmessage.json`,
+	}
+
+	actual := make(map[string]string)
+	populateTemplates(embedTest, actual, "testdata")
+
+	if !reflect.DeepEqual(actual, expected) {
+		t.Fatalf("expected templates: %v, actual: %v", expected, actual)
+	}
+}
+
+func TestGetAsDataTemplates(t *testing.T) {
+	testData := make(map[string]string)
+	populateTemplates(embedTest, testData, "testdata")
+
+	tests := []struct {
+		name     string
+		input    map[string]string
+		expected []data.Template
+	}{
+		{
+			name:  "testdata",
+			input: testData,
+			expected: []data.Template{
+				{
+					Name:        "test",
+					RegoPackage: "test.rawmessage.json",
+				},
+			},
+		},
+		{
+			name:  "testdata",
+			input: EmbeddedCommon(),
+			expected: []data.Template{
+				{
+					Name:        "common",
+					RegoPackage: "postee",
+				},
+			},
+		},
+		{
+			name:     "empty",
+			input:    map[string]string{},
+			expected: []data.Template{},
+		},
+		{
+			name: "bad rego",
+			input: map[string]string{
+				"test.test": "packageconnected",
+			},
+			expected: []data.Template{},
+		},
+		{
+			name: "bad rego 2",
+			input: map[string]string{
+				"test.test": "package ",
+			},
+			expected: []data.Template{},
+		},
+		{
+			name: "bad rego 3",
+			input: map[string]string{
+				"test.test": "package",
+			},
+			expected: []data.Template{},
+		},
+		{
+			name: "empty rego",
+			input: map[string]string{
+				"test.test": "",
+			},
+			expected: []data.Template{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actual := getAsDataTemplates(test.input)
+			if !reflect.DeepEqual(actual, test.expected) {
```

Comment:
You can use `assert.Equal(t, ...)` for all comparisons you made in this test file. No need to use `reflect` or `==`

Reply:
`assert.Equal ` uses `reflect.DeepEqual` behind the scene, so it doesn't really matter.

Reply:
You are right, there is no error in logic. I just mentioned `assert` just because it makes code shorter and more "unit test friendly" (for cases when errors are reported, .. etc)